### PR TITLE
DEPR: Removing previously deprecated datetools module (#6581)

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] related to #6581
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -314,6 +314,7 @@ Removal of prior version deprecations/changes
 - The ``Panel4D`` and ``PanelND`` classes have been removed (:issue:`13776`)
 - The ``Panel``class has dropped the ``to_long``and ``toLong`` methods (:issue:`19077`)
 - The options ``display.line_with`` and ``display.height`` are removed in favor of ``display.width`` and ``display.max_rows`` respectively (:issue:`4391`, :issue:`19107`)
+- Removal of the previously deprecated module ``pandas.core.datetools`` (:issue:`14105`, :issue:`14094`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: move to 0.24.0

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. though not running on travis for some reason.

**Comment by jreback** [comment]
cc @TomAugspurger 

maybe need to login to travis and turn on pandas-dev for your fork?

**Comment by jbrockmendel** [comment]
@datapythonista needs rebase, hopefully CI problem will fix itself.

**Comment by jreback** [comment]
can you rebase, for some reason travis is not running either - not sure why

**Comment by jreback** [comment]
thanks @datapythonista

**Pull request merged**